scripts/ci/check_compliance.py

```diff
@@ -1447,6 +1447,62 @@ def run(self):
             with open(file, "r") as fp:
                 self.check_file(file, fp)
 
+class FileEncoding(ComplianceTest):
+    """
+    Check the encoding for source files.
+    """
+    name = "FileEncoding"
+    doc = "Check the encoding for source files."
+    path_hint = "<git-top>"
+
+    MATCH_FILES = [
+            r'.*\.c$',
+            r'.*\.h$',
+            r'.*Kconfig.*',
+            r'.*CMakeLists.txt$',
+            ]
```

Comment:
.py? .sh? .yaml?

Reply:
PY hits a ton of stuff and I would leave those out because the language really like to support utf-8, sh and yaml sounds alright

Reply:
added sh and yaml, updated the list on https://gist.github.com/fabiobaltieri/44c3639556bea390860b89b73383f048